### PR TITLE
Use floating versions in Haiku package names

### PR DESCRIPTION
**Pull request by am11**
Per the conversation with @waddlesplash in https://github.com/dotnet/arcade/pull/15091.

### CODE REVIEW

**Comment by am11** [comment]
Tested with `eng/common/cross/build-rootfs.sh haiku x64`

File: eng/common/cross/build-rootfs.sh [hunk 1]
```diff
@@ -91,14 +91,14 @@ __HaikuPackages="gcc_syslibs"
...
 __HaikuPackages+=" krb5_devel"
 __HaikuPackages+=" libiconv"
 __HaikuPackages+=" libiconv_devel"
-__HaikuPackages+=" llvm12_libunwind"
-__HaikuPackages+=" llvm12_libunwind_devel"
+__HaikuPackages+=" llvm[0-9]*_libunwind"
+__HaikuPackages+=" llvm[0-9]*_libunwind_devel"
 __HaikuPackages+=" mpfr"
 __HaikuPackages+=" mpfr_devel"
 __HaikuPackages+=" openssl"
```
Comment (waddlesplash) [inline eng/common/cross/build-rootfs.sh:104]: You should probably also change this to `openssl3` (no need for a regex here)

Comment (waddlesplash) [reply eng/common/cross/build-rootfs.sh:104]: And then `openssl3_devel`.

Comment (am11) [reply eng/common/cross/build-rootfs.sh]: Thanks, I agree that openssl3 is something we should be using from the get-go as linux was upgraded relatively recently to that plan (@trungnt2910 has some port work in progress, so a heads-up).

**Comment by am11** [comment]
@mthalman one more update. 😁

**Review by mthalman — APPROVED** [review]

**Pull request merged**